### ISSUE TITLE
Klipper new menu system : How to remove item from default menu ?

### ISSUE THREAD

**Issue by Arti4ever** [KevinOConnor/klipper#3199]
Hi @mcmatrix,

I just started to migrate from old to new menu system and I don't find how I can remove item from the default menu. 
For example, in the Main menu, how can I do to have only Octoprint and Filament sub-menus ?

With the old menu system, we could redefine the list in each menu but with the new system it seems automatic. 

Can you advise me ?
Thanks

**Comment by klipper-gitissuebot** [comment]
Hi @Arti4ever,

It did not look like there was a Klipper log file attached to this ticket. The log file has been engineered to answer common questions the Klipper developers have about the software and its environment (software version, hardware type, configuration, event timing, and hundreds of other questions).

Unfortunately, too many people have opened tickets without providing the log. That consumes developer time; time that would be better spent enhancing the software. If this ticket references an event that has occurred while running the software then the Klipper log must be attached to this ticket. Otherwise, this ticket will be automatically closed in a few days.

For information on obtaining the Klipper log file see: https://github.com/KevinOConnor/klipper/blob/master/docs/Contact.md

The log can still be attached to this ticket - just add a comment and attach the log to that comment.

Best regards,
~ Your friendly GitIssueBot

PS: I'm just an automated script, not a human being.

**Comment by mcmatrix** [comment]
Hi @Arti4ever,

Yes, this new menu has a different method for building the menu hierarchy.
It doesn't have manual `items` attribute anymore.
The menu hierarchy is built automatically from the config sections:
```ini
[menu __main]
...
[menu __main __control]
...
[menu __main __control __home]
...
etc
```

I think you can try to override and disable (by setting `enable: False`) top-level list items you don't want to see.
I haven't tried it.
```ini
[menu __main __control]
type: list
enable: False
name: Control
```

or you can define your own root level menu and submenus.
Something like this
```ini
[menu __mymenu]
...
[menu __mymenu __abcd1]
...
[menu __mymenu __abcd2]
...
```

and then under `display` section set custom root menu for the menu system.
`menu_root: __mymenu`

**Comment by Arti4ever** [comment]
Thank you @mcmatrix ! 
First method works great.

**Comment by th33xitus** [comment]
@mcmatrix I just stumbled across the same question and your solution of changing the menu_root made it possible to create a new, seperate menu. I just wanted to inform you that this parameter you mentioned... the `menu_root: __mymenu` is not documented anywhere. Or did i just not find it? If it's true that it isn't documented, one should consider to do so. I found it really important and especially a very simple solution to deactivate the default menu and make klipper load my custom menu.
